lib/config/error-codes/throw.js

```diff
@@ -72,25 +84,84 @@ function throwError (domain, subdomain, error, ...placeholders) {
   throw getError(domain, subdomain, error, ...placeholders);
 }
 
+
 /**
- * Wrap getError and throw with the provided domain and subdomain.
+ * Construct and return the corresponding error
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function getErrorFrom (errorSource, domain, subdomain, error, ...placeholders) {
```

Comment:
Instead of re-implementing everything, this funciton should only call `getError` and then inject the stacktrace from the origin error into the new one.

Reply:
If I do that I will have to change the signature of `getError` like function `getErrorFrom (errorSource, domain, subdomain, error, ...placeholders)` and change the call of `getError` everywhere like `errorsManager.getError(null, `api`, `admin`, 'database_not_found', database);`
Or if it's called with the `wrap` method `errorsManager.getError(null, 'database_not_found', database);`
So basically if `getError` handles every cases, it can become a little bit confusing, don't you think ?

Reply:
Hm... that's not what I'm suggesting.

What I said is that you should implement a `getErrorFrom` like this:

```suggestion
function getErrorFrom (errorSource, domain, subdomain, error, ...placeholders) {
  const error = this.getError(domain, subdomain, error, ...placeholders);
  
  error.stack = errorSource.stack;
  
  return error;
}
```

Reply:
... -_- Right.
Sorry misunderstanding.